### PR TITLE
Pass along auth token for kernel gateway

### PR DESCRIPTION
**Pull request by jhpedemonte**
Ref #20

### CODE REVIEW

**Comment by jhpedemonte** [comment]
Only tested with kernel gateway. Will test with tmpnb.

File: routes/api.js [hunk 1]
```diff
@@ -10,13 +10,14 @@ var config = require('../app/config');
 var Promise = require('es6-promise').Promise;
 var url = require('url');
 
-var KERNEL_GATEWAY_URL = config.get('KERNEL_GATEWAY_URL');
+var kgUrl = config.get('KERNEL_GATEWAY_URL');
+var kgAuthToken = config.get('KG_AUTH_TOKEN');
```
Comment (parente) [inline routes/api.js]: Set a false-y default (blank string or null) in `config.json` which is serving as documentation of all of the defaults and optional settings when you run `make help`.

Comment (jhpedemonte) [inline routes/api.js]: Done.

**Comment by parente** [comment]
I'll have a look at this once it's rebased.

**Comment by jhpedemonte** [comment]
OK, got things working again after the merge. Might be able to do further cleanup and remove duplication of code with a `define` or two. Suggestions welcome.

File: Makefile [hunk 2]
```diff
@@ -167,21 +172,43 @@ run-tmpnb-pool: token-check
...
+integration-test: IP?=$$(docker-machine ip $$(docker-machine active))
+integration-test: KG_PORT?=8888
+integration-test: KG_AUTH_TOKEN?=1a2b3c4d5e6f
+integration-test: | kill build run-kernel-gateway
+	$(DOCKER_APP) -d \
+		-e KERNEL_GATEWAY_URL=http://$(KG_CONTAINER_NAME):8888 \
+		-e KG_AUTH_TOKEN=$(KG_AUTH_TOKEN) \
+		--link $(KG_CONTAINER_NAME):$(KG_CONTAINER_NAME) \
+		$(DASHBOARD_IMAGE_NAME)
+	@echo 'Waiting 30 seconds for server to start...'
```
Comment (parente) [inline Makefile]: Can probably shorten this quite a bit because there's another docker build that happens right after it.

**Pull request merged**